system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.hpp

```diff
@@ -107,8 +99,34 @@ class PeriodicMeasurementNode : public system_metrics_collector::Collector, publ
    * If true, then measurements are cleared after publishing data
    */
   const bool clear_measurements_on_publish_;
+
+  rclcpp::Time measurement_start_;
   rclcpp::TimerBase::SharedPtr measurement_timer_;
   rclcpp::TimerBase::SharedPtr publish_timer_;
+
+protected:
+  /**
+   * Creates a ROS2 timer with a period of measurement_period_.
+   *
+   * @return if setup was successful
+   */
+  bool setupStart() override;
+
+  /**
+   * Stops the ROS2 timer
+   *
+   * @return if teardown was successful
+   */
+  bool setupStop() override;
+
+  /**
+   * Returns a new MetricsMessage with some common information filled out
+   *
+   * @return new MetricsMessage
+   */
+  metrics_statistics_msgs::msg::MetricsMessage newMetricsMessage();
```

Comment:
```suggestion
  metrics_statistics_msgs::msg::MetricsMessage newMetricsMessage() const;
```

Reply:
The relevant [change](https://github.com/ros2/rclcpp/pull/922) to rclcpp seems to have been merged. I and other developers and will need to update our development environment. Does the CI always build off of the latest ROS2 master?